zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/metrics/UsageMetricsExportProcessor.java

```diff
@@ -90,15 +93,57 @@ public void processRecord(final TypedRecord<UsageMetricRecord> record) {
             .setResetTime(record.getTimestamp());
 
     final var bucket = usageMetricState.getActiveBucket();
-    if (bucket != null && !bucket.getTenantRPIMap().isEmpty()) {
-      eventRecord
-          .setEventType(EventType.RPI)
-          .setStartTime(bucket.getFromTime())
-          .setEndTime(bucket.getToTime())
-          .setValues(bucket.getTenantRPIMapValue());
+    if (bucket != null) {
+      processMetricType(
+          bucket,
+          eventRecord,
+          EventType.RPI,
+          bucket.getTenantRPIMap(),
+          bucket.getTenantRPIMapValue());
+      processMetricType(
+          bucket,
+          eventRecord,
+          EventType.EDI,
+          bucket.getTenantEDIMap(),
+          bucket.getTenantEDIMapValue());
+    } else {
+      checkRecordLength(eventRecord).forEach(this::appendFollowUpEvent);
     }
+  }
+
+  /** Processes a specific metric type and appends the resulting records. */
+  private void processMetricType(
+      final PersistedUsageMetrics bucket,
+      final UsageMetricRecord baseRecord,
+      final EventType eventType,
+      final Map<?, ?> conditionMap,
+      final DirectBuffer valuesBuffer) {
+    if (!conditionMap.isEmpty()) {
+      final UsageMetricRecord clonedRecord = cloneEventRecord(baseRecord);
+      enhanceEventRecord(clonedRecord, bucket, eventType, valuesBuffer);
+      checkRecordLength(clonedRecord).forEach(this::appendFollowUpEvent);
+    }
+  }
 
-    checkRecordLength(eventRecord).forEach(this::appendFollowUpEvent);
+  /** Creates a deep clone of the given UsageMetricRecord. */
+  private UsageMetricRecord cloneEventRecord(final UsageMetricRecord original) {
```

Comment:
🙃 Technically it's not a clone because we only copy 3 properties. Maybe more like `initializeEventRecord`?